### DISCUSSION TITLE
Slim Select 2.0!!!

### DISCUSSION THREAD

**Discussion by brianvoe** [brianvoe/slim-select#356] [Announcements]
Version 2 release today! 

The biggest part of this release was completely restructuring communication between javascript classes. The structure now is based upon responsibility. So the main index will communicate with the store, original select and rendering as it sees fit. If the original select changes, it will communicate that back to the main index and based upon that callback will choose what to do with that information. Same goes for the render for searching and option selection.

The next decent change is the separation of settings from events into sub objects. This makes it easier for a couple of different reasons. 1. Reusing certain settings but having different change callbacks. 2. Next step is I am going to release a vue version and passing props is easier as an object of fields appose to individual fields. I would assume that to be the case for react as well, which I do plan on releasing(might need some help from the community). This does require migration of old code into the correct sub object of settings.

With the settings and events separation some of the settings fields and event callback functions may have some slightly different names or return values. This was to help clean things up. You shouldn't have any missing features from v1 they may just be handled slightly differently.

With that being said the last change was with getting rid of ajax. Well sort of. Slim Select 2 now deals with these ajax/fetch based events as promise returns in the search event. See docs for more detail.

Slim select was fully built with typescript. This helped tremendously for finding issues not easily catch able without typescript declarations. the dist output has all the declarations for slim select in it.

Hopefully this gets everyone back on track with this project and if there is anything that needs address please file an issue and if you can please try to create a pr for it. Hopefully its easier to follow and understand now. Thanks!

<hr /><em>This discussion was created from the release <a href='https://github.com/brianvoe/slim-select/releases/tag/v2.0.0'>Slim Select 2.0!!!</a>.</em>

**Comment by polishchukI** [comment]
Hello! Is there any manual, to migrate from 1.27.1=>final version? I have a lot of issues vhile I try to change only js and css files((

**Comment by brianvoe** [comment]
Sorry i dont. You shouldnt see anything crazy maybe some minor naming changes. I upgraded this to 2.0 so long ago i dont fully remember everything i changed. Check out the docs, everything you did previously should be pretty similar.

**Reply by polishchukI** [reply]
Where can i find documentation for 1.27.1? for making live search in modal.

I need to make live search direct in sql for product.

**Reply by dkniffin** [reply]
+1 for needing access to 1.x docs. I'm currently upgrading to 2.x, but it's hard to tell what's changed, especially when the v1 docs are no longer accessible.'

Edit: it looks like the doc source code is still available in here: https://github.com/brianvoe/slim-select/tree/v1.27.1/src/docs/pages

**Reply by brianvoe** [reply]
Sorry its hard enough to get time to work on things but its been 2 years since i released v2. I cant justify putting time into that.

**Reply by dkniffin** [reply]
I just finished updating my application to use v2. This is not a comprehensive list by any means, but these were the biggest changes I noticed while upgrading, in case anyone else needs them:

- `.ss-content` element is now in a separate part of the DOM
- `.ss-single-selected` and `.ss-multi-selected` elements were removed
- `.placeholder` class was renamed to `.ss-placeholder`
- `.ss-add` element was removed
- Many of the JS options were moved into the `settings` and `events` sub-keys
- `.set` method was renamed to `.setSelected`
- `placeholder` setting was renamed to `placeholderText`
- `showContent` setting was renamed to `openPosition`
- `innerHtml` setting was renamed to `html`
- `deselectLabel` setting was removed
- `data-ssid="*"` HTML attribute was renamed to `data-id="ss-*"` (and therefore the workaround for [this issue](https://github.com/brianvoe/slim-select/issues/273) has changed)
- [A bunch of CSS variables](https://github.com/brianvoe/slim-select/blob/373c54395c7289cb3bbc8bfa92f7ec8ed9d49591/src/slim-select/slimselect.scss#L3-L25) were added, which can be used to configure colors, sizes, etc

**Reply by brianvoe** [reply]
Thats a pretty good list. Nice job. Hopefully the 2nd version of my package is much better for you.

**Comment by polishchukI** [comment]
I have issue with responce from server, it appending it to select, but it is invisible on page, and select stays blank and can not be selected

Получите Outlook для Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by brianvoe** [reply]
pull an old tag. npm install, npm run dev, localhost:8080

**Comment by yukulele** [comment]


**Reply by brianvoe** [reply]
sorry not a thing
